<a href="https://colab.research.google.com/github/Sameeulla-A/PyTorch/blob/main/2_Math_and_Printing.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [6]:
x = 7
y = 4.3
z = 0

In [7]:
y + x -7

4.300000000000001

if we use to do the above calc we will get 4.3. but here we are getting 4.300000000000001 some values extra. shorter version of the explation is floating point algebra is not same thing as real algebra. here we are not actully implementing real algebra, it's called floating point algebra. because computers have limited precision, they cannot represent arbitratily small or arbitrarlily large numbers. so we get some tiny, tiny computer rounding errors they are essentially zeros.

In [ ]:
c = y + 3.5*x - 7
c

21.8

In [ ]:
4/3

1.3333333333333333

In [ ]:
4\
  3
# \ indicating to python to continue the line of code into the next line of text.

SyntaxError: invalid syntax (2376776811.py, line 2)

this is two lines of text, but it's one line of code, \ treated as space in python

In [ ]:
4 \
  + 3 # it would be a meaning ful

7

In [ ]:
3*2 # multiplication

6

In [ ]:
 9**2  #root


81

In [ ]:
9**1/2

4.5

it should be equal to 3, but python follows the order of operations. python interpreting (9**1)/2. let's reorder the same.

In [ ]:
9**(1/2) # 9 power of the half

3.0

In [2]:
Firstname = 'mike'
lastname = 'cohen'

Firstname + lastname

'mikecohen'

+ sign can be used concatenation on strings, this is a phenomenon in codeing called overloading.
overloading is a term when you have the same operation that's used in different ways depending on the context. so this plus sign means to do mathematical addition with number and concatenation with strings.

In [3]:
fullname = Firstname + ' ' + lastname  # ' ' used for space
fullname

'mike cohen'

In [1]:
# let's do a substraction
Firstname - lastname

NameError: name 'Firstname' is not defined

python know what to with + and it doesn't know what to do with - while working with Strings

In [4]:
Firstname*3   # prints 3 times

'mikemikemike'

In [5]:
print('7*3 = 21')
print('7*3 = 2.1')

7*3 = 21
7*3 = 2.1


in the above we are not doing math, it's a string thing and we are printing the strings not numbers.

In [10]:
print(x+y)  # here variables used without quotes.
print('x+y') #here variables used within the quotes, so python treated as string.

11.3
x+y


In [11]:
print(x + ' + ' + y + ' = ' + x+y) # here we are trying to combine strings and Ints

TypeError: unsupported operand type(s) for +: 'int' and 'str'

In [12]:
print(str(x) + ' + ' + str(y) + ' = ' + str(x+y))

7 + 4.3 = 11.3


in the above we used str() funtion to convert the int / float into string. it's a complecated code, visually.

In [14]:
print(f' x + ' + ' + y + ' == ' + x+y')

False


input() function allows user to input the data.

In [15]:
ans = input('Give Me a Number: ')

Give Me a Number: 8


In [17]:
ans # gives the output in ''s means it's a string.

'8'

In [18]:
type(ans)

str

In [16]:
ans + 8 #string + int doesn't support.

TypeError: can only concatenate str (not "int") to str

In [19]:
int(ans) + 8   #converted into integer

16

In [20]:
float(ans) + 7 # converted into float

15.0

In [21]:
qns = float( input('Give me a number: '))
type(qns)

Give me a number: 15.2


float

In [ ]:
# Exercise!
# Lets test the Pythagorean Theorem.
# c^2 = a^2 + b^2
# c**2 = a**2 + b**2

# get two input from the user
#   side 'a'
#   side 'b'
# - compute and print out the length of side c!

In [22]:
a = float(input('side a: '))
b = float(input('side b: '))
c = a**2 + b**2
print(f'side c: {c}')

side a: 15.2
side b: 15.2
side c: 462.08


code worked but, not exactly as theorem

In [23]:
a = float(input('side a: '))
b = float(input('side b: '))
c = (a**2 + b**2)**(1/2)
print('side c is ' + str(c))

side a: 15.2
side b: 15.2
side c is 21.496046148071045
